test/unit/org/apache/cassandra/service/reads/range/ReplicaPlanMergerTest.java

```diff
@@ -0,0 +1,460 @@
+                       range(min(), max(10)),
+                       range(max(10), max(20)),
+                       range(max(20), max(30)),
+                       range(max(30), max(40)),
+                       range(max(40), min()));
+            testRanges(cl,
+                       range(min(), max(25)),
+                       range(min(), max(10)),
+                       range(max(10), max(20)),
+                       range(max(20), max(25)));
+            testRanges(cl,
+                       range(min(), max(40)),
+                       range(min(), max(10)),
+                       range(max(10), max(20)),
+                       range(max(20), max(30)),
+                       range(max(30), max(40)));
+            testRanges(cl,
+                       range(min(), max(50)),
+                       range(min(), max(10)),
+                       range(max(10), max(20)),
+                       range(max(20), max(30)),
+                       range(max(30), max(40)),
+                       range(max(40), max(50)));
+            testRanges(cl,
+                       range(max(20), max(30)));
+            testRanges(cl,
+                       range(max(25), min()),
+                       range(max(25), max(30)),
+                       range(max(30), max(40)),
+                       range(max(40), min()));
+            testRanges(cl,
+                       range(max(50), min()));
+            testRanges(cl,
+                       range(max(30), max(10)),
+                       range(max(30), max(40)),
+                       range(max(40), min()),
+                       range(min(), max(10))); // wrapped
+            testRanges(cl,
+                       range(max(25), max(15)),
+                       range(max(25), max(30)),
+                       range(max(30), max(40)),
+                       range(max(40), min()),
+                       range(min(), max(10)),
+                       range(max(10), max(15))); // wrapped
+        }
+    }
+
+    /**
+     * Tests range merging with a multinode cluster when the token ranges don't overlap between replicas.
+     */
+    @Test
+    public void testMultiNodeWithContinuousRanges()
+    {
+        new TokenUpdater().withTokens("127.0.0.1", 10, 20, 30)
+                          .withTokens("127.0.0.2", 40, 50, 60)
+                          .withTokens("127.0.0.3", 70, 80, 90)
+                          .update();
+
+        // with CL=ANY the ranges are fully merged (unless they are wrapping)
+        testMultiNodeFullMerge(ANY);
+
+        // with CL=THREE the ranges are not merged at all
+        testMultiNodeNoMerge(THREE);
+
+        // with CLs requiring a single node the ranges are merged in a per-node basis
+        for (ConsistencyLevel cl : Arrays.asList(ONE, LOCAL_ONE))
+        {
+            testRanges(cl,
+                       range(min(), min()),
+                       range(min(), max(60)),
+                       range(max(60), min()));
```

Comment:
nit: I know it's not always possible, but if the `expected` ranges fit on one line, that helps visually distinguish them from the query range (rather than having to remember that the first one is the query range).

Reply:
Done [here](https://github.com/apache/cassandra/pull/801/commits/3fd6f859b42a8934b1ee8dd46342f9b3637fc164).